### PR TITLE
various Solidity Foundry updates

### PR DESCRIPTION
**Pull request by Tofel**
This PR introduces a couple of changes:
* adds two new jobs that gather result of matrix jobs for tests and formatting, so that we can use them as required checks
* modifies formatting job condition in such a way that it also runs for test files
* adds back static analysis with Slither
* post Solidity Review Ticket comment only if such Jira issue was found/created

Tests:
- [x] if all test jobs are skipped, check job passes
- [x] if any of the test jobs fails, check job fails
- [x] if any of the test jobs is cancelled, check job fails
- [x] if all format jobs are skipped, check job passes
- [x] if any of the format jobs fails, check job fails
- [x] if any of the format jobs is cancelled, check job fails
- [x] if neither tests nor format jobs run, both checks pass
- [x] slither with diff runs only if Sol files are modified
- [x] slither config upload works
- [x] if slither config upload fails, workflow is failed
- [x] formatting jobs run also if test files are modified
- [x] no Solidity Review comment is created, if ticket is not found
- [x] Solidity Review comment is created, if ticket is found

### CODE REVIEW

**Review by gheorghestrimtu — APPROVED** [review]

File: .github/workflows/solidity-foundry.yml [hunk 1]
```diff
@@ -252,9 +252,286 @@ jobs:
           artifact-name: code-coverage-report-${{ matrix.product.name }}
           working-directory: ./contracts
 
+  # runs only if non-test contracts were modified; scoped only to modified or added contracts
+  analyze:
```
Comment (Tofel) [inline .github/workflows/solidity-foundry.yml:256]: I added back whole static analysis job

File: .github/workflows/solidity-foundry.yml [hunk 2]
```diff
@@ -252,9 +252,286 @@ jobs:
...
+      - name: Upload configs
+        if: needs.changes.outputs.sol_mod_only == 'true'
+        uses: actions/upload-artifact@v4.4.3
+        timeout-minutes: 2
+        with:
+          name: tmp-configs-${{ github.sha }}
+          path: contracts/configs
+          retention-days: 7
+          if-no-files-found: error
+          include-hidden-files: true
```
Comment (Tofel) [inline .github/workflows/solidity-foundry.yml:346]: this is the only real change vs previous version, required as config files start with a `.`

File: .github/workflows/solidity-foundry.yml [hunk 3]
```diff
@@ -252,9 +252,286 @@ jobs:
...
+        if: ${{ contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled') }}
+        run: |
+          echo "At least one test job failed or was cancelled. Please check the logs."
+          exit 1
+      - run: echo 'Success'
+
   solidity-forge-fmt:
     name: Forge fmt ${{ matrix.product.name }}
-    if: ${{ needs.changes.outputs.non_src_changes == 'true' || needs.changes.outputs.not_test_sol_modified == 'true' }}
+    if: ${{ needs.changes.outputs.non_src_changes == 'true' || needs.changes.outputs.sol_modified_added == 'true' }}
```
Comment (Tofel) [inline .github/workflows/solidity-foundry.yml:534]: we will now run `forge fmt` when any Solidity files change, not only non-test ones

File: .github/workflows/solidity-foundry.yml [hunk 4]
```diff
@@ -252,9 +252,286 @@ jobs:
...
+            Hey @${{ github.event.push && github.event.push.pusher && github.event.push.pusher.username || github.actor }}, you can view Slither reports in the job summary [here](${{ steps.job-summary-url.outputs.job_summary_url }}) or download them as artifact [here](${{ steps.build-slither-artifact-url.outputs.slither_artifact_url }}).
+            Please check them before merging and make sure you have addressed all issues.
+          edit-mode: replace
+
+      - name: Remove temp artifacts
+        uses: geekyeggo/delete-artifact@24928e75e6e6590170563b8ddae9fac674508aa1 # v5.0
+        with:
+          name: tmp-*
+
+  check-tests-results:
```
Comment (Tofel) [inline .github/workflows/solidity-foundry.yml:519]: new job that checks for results of the test matrix, we will set it as a required job

File: .github/workflows/solidity-foundry.yml [hunk 5]
```diff
@@ -285,3 +562,16 @@ jobs:
         working-directory: contracts
         env:
           FOUNDRY_PROFILE: ${{ matrix.product.name }}
+
+  check-fmt-results:
```
Comment (Tofel) [inline .github/workflows/solidity-foundry.yml:566]: new job that checks for results of the formatting matrix, we will set it as a required job

**Review by momentmaker — APPROVED** [review]

File: .github/workflows/solidity-tracability.yml
Comment (FelixFan1992) [inline .github/workflows/solidity-tracability.yml:1]: typo `traceability`

Comment (FelixFan1992) [reply .github/workflows/solidity-tracability.yml:1]: also the name at line 2

**Pull request merged**